src/Symfony/Component/VarDumper/Caster/DsCaster.php

```diff
@@ -27,23 +23,48 @@
  */
 class DsCaster
 {
-    /**
-     * @param Set|Deque|Vector|Stack|Queue|PriorityQueue $c
-     */
-    public static function castDs($c, array $a, Stub $stub, bool $isNested): array
+    public static function castCollection(Collection $c, array $a, Stub $stub, bool $isNested): array
     {
-        $prefix = Caster::PREFIX_VIRTUAL;
-        $a = $c->toArray();
-        $a[$prefix.'capacity'] = $c->capacity();
+        $a[Caster::PREFIX_VIRTUAL.'count'] = $c->count();
+        $a[Caster::PREFIX_VIRTUAL.'capacity'] = $c->capacity();
+
+        if (!$c instanceof Map) {
+            foreach ($c as $k => $v) {
+                $a[Caster::PREFIX_VIRTUAL.$k] = $v;
```

Comment:
why `Caster::PREFIX_VIRTUAL`? If I add my own class implementing the `Collection` interface and have a `count` or `capacity` key there then this would overwrite those values.

Reply:
Why would you do that? Using no prefix means "dynamic property", which is semantically wrong.
I don't think we really care.

Reply:
"dynamic property" seems correct to me in the context of data structure keys - they are not actual properties of the structure and are dynamic

Reply:
here is what a dynamic property is:
```php
$map = new Ds\Map();
$map->abc = 123;
```

Reply:
but anyway, I don't really care, so here you are :)

Reply:
oh... yeah you're correct I forgot that PHP allows this... Still who would do it? In the end both scenarios are equally unlikely so just choose whichever you feel is better.

Reply:
as is now works for me

Reply:
Perhaps check how it's done for things like Spl structures or ArrayObject and do it consistently?

Reply:
They do it by nesting in a "storage" virtual property, as var_dump does for them. Let's keep as is now that's fine.